src/tests/test_combine_latest.cpp

```diff
@@ -0,0 +1,155 @@
+
+        rpp::source::just(rpp::schedulers::trampoline{}, 1, 2, 3)                      // source 1
+            .combine_latest(rpp::source::just(rpp::schedulers::trampoline{}, 4, 5, 6)) // source 2
+            .subscribe(mock);
+
+        // Above stream should output in such sequence
+        // source 1:   -1---2---3-|
+        // source 2: -4---5---6-|
+
+        CHECK(mock.get_received_values() == std::vector<std::tuple<int, int>>{
+            std::make_tuple(1, 4),
+            std::make_tuple(1, 5),
+            std::make_tuple(2, 5),
+            std::make_tuple(2, 6),
+            std::make_tuple(3, 6),
+        });
+        CHECK(mock.get_on_completed_count() == 1);
+        CHECK(mock.get_on_error_count() == 0);
+    }
+
+    GIVEN("observable of -1-2-3-| combines with two other sources on trampoline")
+    {
+        auto mock = mock_observer<std::tuple<int, int, int>>{};
+
+        rpp::source::just(rpp::schedulers::trampoline{}, 1, 2, 3)          // source 1
+            .combine_latest(
+                rpp::source::just(rpp::schedulers::trampoline{}, 4, 5, 6), // source 2
+                rpp::source::just(rpp::schedulers::trampoline{}, 7, 8 ,9)) // source 3
+            .subscribe(mock);
+
+        // Above stream should output in such sequence
+        // source 1:   --1---2---3-|
+        // source 2: -4---5---6-|
+        // source 3: --7---8---9-|
+
+        CHECK(mock.get_received_values() == std::vector<std::tuple<int, int, int>>{
+            std::make_tuple(1, 4, 7),
+            std::make_tuple(1, 5, 7),
+            std::make_tuple(1, 5, 8),
+            std::make_tuple(2, 5, 8),
+            std::make_tuple(2, 6, 8),
+            std::make_tuple(2, 6, 9),
+            std::make_tuple(3, 6, 9),
+        });
+        CHECK(mock.get_on_completed_count() == 1);
+        CHECK(mock.get_on_error_count() == 0);
+    }
+}
+
+SCENARIO("combine_latest defers completed event", "[combine_latest]")
+{
+    GIVEN("observable of -1-2-3-| and never")
+    {
+        auto mock = mock_observer<std::tuple<int, int>>{};
+        rpp::source::just(1, 2, 3)
+            .combine_latest(rpp::source::never<int>())
+            .subscribe(mock);
+
+        CHECK(mock.get_received_values().empty());
+        CHECK(mock.get_on_completed_count() == 0);
+        CHECK(mock.get_on_error_count() == 0);
+    }
+}
+
+SCENARIO("combine_latest forwards errors", "[combine_latest]")
+{
+    GIVEN("observable of -1-2-3-| combines with error")
+    {
+        auto mock = mock_observer<std::tuple<int, int>>{};
+        rpp::source::just(1, 2, 3)
+            .combine_latest(rpp::source::error<int>(std::make_exception_ptr(std::runtime_error{""})))
+            .subscribe(mock);
+
+        CHECK(mock.get_received_values().empty());
+        CHECK(mock.get_on_completed_count() == 0);
+        CHECK(mock.get_on_error_count() == 1);
+    }
+}
+
+SCENARIO("combine_latest regards unsubscription", "[combine_latest]")
```

Comment:
These tests uselsess, because subscribe with unsubscribed observable do nothing [link](https://github.com/victimsnino/ReactivePlusPlus/blob/b87592fda89d76ac203ab6359d424dad9934fafa/src/rpp/rpp/observables/specific_observable.hpp#L63)